daffodil-macro-lib/src/main/scala/org/apache/daffodil/util/TimerMacros.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.util
+
+import scala.reflect.macros.blackbox.Context
+
+object TimeTrackerMacros {
+
+  def trackMacro(c: Context)(name: c.Tree)(body: c.Tree) = {
```

Comment:
The usage of childrenTimeStack could use a comment.

My first reading was that each block had its own childrenTimeStack, that each of the children pushed onto. This seems to be backwards to what is actually happening, where there is a single stack, where the top element is the total time that all of the children parsers took. What is actually happening is reasonable (and seems like the correct thing to do), but a comment explaining it would be nice.

Reply:
Yeah, your reading of the code is correct. I'll add a coment to that affect and update our performance page to describe how one might use this new timer.